src/leader_election.rs

```diff
@@ -276,7 +307,9 @@ pub mod ballot_leader_election {
             ));
         }
 
-        fn handle_reply(&mut self, rep: HeartbeatReply) {
+        fn handle_reply(&mut self, from: u64, rep: HeartbeatReply) {
+            debug!(self.logger, "Heartbeat reply from {}", from);
+
             if rep.round == self.hb_round {
                 self.ballots.push((rep.ballot, rep.majority_connected));
             } else {
```

Comment:
log with `warn!` if we get a late response. It should log the current round, the ballot and `self.hb_current_delay`